### PR TITLE
Feat erasure coding

### PR DESCRIPTION
**Pull request by ChouaibHamek**
Created the two methods for file shredding and file recovery from a buffer that contains some of the shreds (respecting the parity used during the shredding).

### CODE REVIEW

File: app/utils/file.js [hunk 1]
```diff
@@ -29,11 +31,79 @@ function decrypt(file, key) {
...
+    shardSize,
+    (error) => {
+      if (error) throw error;
+      // Parity shards now contain parity data.
+    }
+  );
+  // writing data shards as files
+  for (let i = 0; i < totalShards; i += i) {
+    // Generate shred IDs to name the shreds
+    fs.writeFileSync(`${i}_${Math.random()}`, buffer.slice(i * shardLength, (i + 1) * shardLength));
```
Comment (yehiahesham) [inline app/utils/file.js:129]: Probaly you made randomizor for now, but in general this should be the UNIQUE Shred_ID supplied by the server (or other ways )  to achieve uniqunes of shreds in the system. I believe we should have a procedure before it to get a list of unique Shred IDs.

Comment (ChouaibHamek) [reply app/utils/file.js:129]: Yes, the shred ID generation is a separate feature. When implemented, we will replace the random naming with the new IDs.

File: app/utils/file.js [hunk 2]
```diff
@@ -29,11 +31,79 @@ function decrypt(file, key) {
...
+  }
+}
+
+// shredsBuffer: a Buffer containing the shreds to be recovered,
+// targets: a variable containing the indecies of the missing shreds
+// dataShreds: Math.ceil(dataBuffer.length / shardLength)
+// recoverdFile: the name of the file to be recovered
+function recoverFile(shredsBuffer, targets, parity, shredLength, dataShreds, recoveredFile) {
+  const buffer = new Buffer(shredsBuffer);
```
Comment (yehiahesham) [inline app/utils/file.js:138]: here you are reading shredsbuffer as a  buffer, that means you will have  a method that reads files(shreds) and concatenate them to a big buffer and call method recoverFile(..) , right ?

Comment (ChouaibHamek) [reply app/utils/file.js:138]: Yes, exactly

**Pull request merged**